views/_layout.njk

```diff
@@ -70,27 +68,77 @@
+                <strong>Success:</strong> {{message}}
+              </div>
+            </div>
+          </div>
+        </div> <!-- /alerts container -->
+      {% endblock %}
 
     <div class="container">
-      <!-- Add body content from child in this block -->
-      {% block content %}{% endblock %}
+      <!-- Add body from child in this "content" block -->
+      {% block content %}
+      {% endblock %}
     </div> <!-- /container -->
 
     <!-- Footer will contain copyright and/or licence information etc. -->
     {% block footer %}
-        <footer class="footer fixed-bottom navbar-dark bg-dark">
-            <div class="container">
-                <p class="text-muted text-strong">Copyright (c) information and other stuff</p>
-            </div>
-        </footer>
+      <footer class="footer fixed-bottom navbar-dark bg-dark">
+        <div class="container">
+          <p class="text-muted text-strong">Copyright (c) information and other stuff</p>
+        </div>
+      </footer>
     {% endblock %}
     
     {% block scripts %}
-    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
-    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js" integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4" crossorigin="anonymous"></script>
+      <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+      <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+      <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js" integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4" crossorigin="anonymous"></script>
+    {% endblock %}
+    
+    {% block alertScripts %}
+      <script>
+        //messagePassing function shows alert message that is passed into DOM from router.
+        //  User can manually dismiss or will auto close after 3 seconds.
+        //parameters: elementId - id of the html page element to be shown/hidden.
+        //            closeId   - id of the button element exposed to user to manually dismiss message.
+        function messagePassing (elementId, closeId) {
+          $(elementId).fadeIn("slow");
+
+          setTimeout(function () {
+            $(elementId).fadeOut("slow");
+          }, 3000);
+
+          $(clodeId).click(function () {
+            $(elementId).fadeOut("slow");
+          });
+        }
+      </script>
+
```

Comment:
This is where the magic happens. If the router passed in a "messageType" element of "error" or "success", the appropriate elements will fade in and fade out via the "messagePassing" function I defined above. So any page/route can trigger this functionality by simply passing the DOM these two elements. 

If desired, we can define/add other "mesageType"s down the road, but I think these two will suffice for most circumstances.